applications/contrastive_phenotyping/contrastive_cli/plot_embeddings.py

```diff
@@ -0,0 +1,144 @@
+# %%
+from pathlib import Path
+
+import pandas as pd
+import plotly.express as px
+import seaborn as sns
+from sklearn.preprocessing import StandardScaler
+from umap import UMAP
+import numpy as np
+from viscy.light.embedding_writer import read_embedding_dataset
+
+# %%
+dataset = read_embedding_dataset(
+    "/hpc/projects/intracellular_dashboard/viral-sensor/infection_classification/models/contrastive_tune_augmentations/predict/2024_02_04-tokenized-drop_path_0_0.zarr"
+)
+dataset
+
+# %%
+# load all unprojected features:
+features = dataset["features"]
+# or select a well:
+# features = features[features["fov_name"].str.contains("B/4")]
+features
+
+# %%
+# examine raw features
+random_samples = np.random.randint(0, dataset.sizes["sample"], 700)
+# concatenate fov_name, track_id, and t to create a unique sample identifier
+sample_id = [
+    str(dataset["fov_name"][idx].values)
+    + "/"
+    + str(dataset["track_id"][idx].values)
+    + "_"
+    + str(dataset["t"][idx].values)
+    for idx in random_samples
+]
+px.imshow(
+    features.values[random_samples],
+    labels={
+        "x": "feature",
+        "y": "sample",
+        "color": "value",
+    },  # change labels to match our metadata
+    y=sample_id,
+    # show fov_name as y-axis
+)
+
+# %%
+scaled_features = StandardScaler().fit_transform(features.values)
+
```

Comment:
What is the rationale for this scaling? We are noticing in the raw embeddings that the cell division events have different mean/std. We should preserve that difference.

Reply:
I took this from the library's [example](https://umap-learn.readthedocs.io/en/latest/basic_usage.html). If I understand correctly this normalize per feature, not per sample. So for example a feature has 10x lower variance in all samples, it will bring it to the same importance as other features. So if a sample have different mean/std across all features, it should be preserved.